featuretools/tests/synthesis/test_deep_feature_synthesis.py

```diff
@@ -1063,58 +1068,67 @@ def test_primitive_options_with_globals(es):
         # ignoring cohorts means no features are created
         assert not isinstance(f.primitive, Month)
 
-        deps = f.get_dependencies()
+        deps = f.get_dependencies(deep=True)
         entities = [d.entity.id for d in deps]
-        identities = [d for d in deps if isinstance(d, IdentityFeature)]
-        variables = [d.variable.id for d in identities]
+        variables = [d for d in deps if isinstance(d, IdentityFeature)]
         if isinstance(f.primitive, Mode):
             assert 'sessions' in entities or 'customers' in entities
```

Comment:
should this check that all values in entities are either 'session' or 'customers'